docs/ExternalResources.md

```diff
@@ -0,0 +1,164 @@
+  - Common utilities and compiler flags for debugging different stages.
+  - Common build configurations.
+  - Tips and tricks for testing: early exiting, filtering tests,
+    testing Swift and Objective-C together, using `PrettyStackTrace`.
+  - A live demo fixing a crash:
+    - Copying program arguments into Xcode for debugging.
+    - Using `dump()` for print debugging.
+- [Getting Started with Swift Compiler Development][] by Brian Gesiak (blog post
+  series, Aug 2017 - Jun 2018)
+- [Contributing to Open Source Swift][] by Jesse Squires (talk, Mar 2016):
+  Covers the following topics:
+  - The overall compiler pipeline.
+  - The organization of different projects part of the Swift monorepo,
+    including some "difficulty levels" for different projects.
+  - Tips for contributing effectively.
+
+[Steps for first PR and asking questions]:
+https://forums.swift.org/t/getting-started-with-swift-compiler-development/31502/2
+[Contributing to Swift for the First Time]: https://youtu.be/51j7TrFNKiA
+[Contributing to Swift compiler]: https://youtu.be/HAXJsgYniqE
+[Contributing to Swift - Part 1]: https://medium.com/kinandcartacreated/contributing-to-swift-part-1-ea19108a2a54
+[Contributing to Swift - Part 2]:
+https://medium.com/kinandcartacreated/contributing-to-swift-part-2-efebcf7b6c93
+[Setting up a compile-edit-test cycle]: https://forums.swift.org/t/need-a-workflow-advice/12536/14
+[Becoming an Effective Contributor to Swift]: https://youtu.be/oGJKsp-pZPk
+[Getting Started with Swift Compiler Development]: https://modocache.io/getting-started-with-swift-development
+[Contributing to Open Source Swift]: https://youtu.be/Ysa2n8ZX-YY
+
+## AST
+
+- [The secret life of types in Swift][] by Slava Pestov (blog post, Jul 2016):
+  This blog post describes the representation of Swift types inside the compiler.
+  It covers many important concepts: `TypeLoc` vs `TypeRepr` vs `Type`, the
+  representation of generic types, substitutions and more.
+  <!-- TODO: It would be great to integrate some of the descriptions
+       in this blog post into the compiler's own doc comments. -->
+
+[The secret life of types in Swift]: https://medium.com/@slavapestov/the-secret-life-of-types-in-swift-ff83c3c000a5
+
+## Type checking and inference
+
+- [Implementing Swift Generics][] by Slava Pestov and John McCall (talk, Oct 2017):
+  This talk dives into how Swift's compilation scheme for generics balances
+  (a) modularity and separate compilation with (b) the ability to pass unboxed
+  values and (c) the flexibility to optionally generate fully specialized code.
+  It covers the following type-checking related topics: type-checking generic
+  contexts, requirement paths and canonicalized generic signatures.
+- [A Type System from Scratch][] by Robert Widmann (talk, Apr 2017):
+  This talk covers several topics related to type-checking and inference in Swift:
+  - Understanding sequent notation which can be used to represent typing judgements.
```

Comment:
Nit: I think we use American English if it's true then this might get changed.
```suggestion
  - Understanding sequent notation which can be used to represent typing judgments.
```

Reply:
🤯 I've always spelt it "judgement", I didn't even know "judgment" was a valid spelling. Turns out "judgment" is the more common spelling, even for British English.